src/writers.scala

```diff
@@ -177,48 +207,51 @@ class LPWriter(writer: Writer) extends LambdaPiWriter(writer)
   {
     c match {
       case Syntax.Rewrite(vars, lhs, rhs) =>
-        val ampvars = vars.map(v => "&" + v)
+        val pat_vars = vars.map(v => "$" + v)
         write("rule ")
-        term(lhs, ampvars)
-        rew
-        term(rhs, ampvars)
+        term(lhs, pat_vars)
+        hook_arrow
+        term(rhs, pat_vars)
       case Syntax.Declaration(id, args, ty, const) =>
+        if (const) write("constant ")
         write("symbol ")
-        if (const) write("const ")
         name(id)
         for (a <- args) { space; block { arg(a) } }
         colon
         term(ty)
       case Syntax.Definition(id, args, ty, tm) =>
-        write("definition ");
+        write("symbol ");
         name(id)
         for (a <- args) { space; block { arg(a) } }
         for (ty <- ty) { colon; term(ty) }
-        dfn
+        colon_equal
         term(tm)
       case Syntax.Theorem(id, args, ty, prf) =>
-        write("theorem ");
+        write("symbol ");
```

Comment:
Are you sure that `symbol` is the right choice here?
Won't this rewrite the theorem name with its proof when used elsewhere?
(This slows down the checker!)

Reply:
Right, "theorem" should be replaced by "opaque symbol".

Reply:
Simply using `symbol` worked, I guess `opaque symbol` will give better performance now